references/detection/train_pytorch.py

```diff
@@ -236,12 +240,18 @@ def main(args):
     train_set = DetectionDataset(
         img_folder=os.path.join(args.train_path, 'images'),
         label_path=os.path.join(args.train_path, 'labels.json'),
-        img_transforms=Compose([
-            T.Resize((args.input_size, args.input_size)),
-            # Augmentations
-            T.RandomApply(T.ColorInversion(), .1),
-            ColorJitter(brightness=0.3, contrast=0.3, saturation=0.3, hue=0.02),
-        ]),
+        img_transforms=Compose(
+            ([T.Resize((args.input_size, args.input_size))] if not args.rotation else [])
+            + [
+                # Augmentations
+                T.RandomApply(T.ColorInversion(), .1),
+                ColorJitter(brightness=0.3, contrast=0.3, saturation=0.3, hue=0.02),
+            ]
+        ),
+        sample_transforms=T.SampleCompose([
+            T.RandomRotate(90, expand=True),
+            T.ImageTransform(T.Resize((args.input_size, args.input_size))),
+        ]) if args.rotation else None
```

Comment:
Would you mind sharing a screenshot in the PR of the output when you add `--show-samples` while using rotation? To make sure the transformation is modifying the text correctly

Reply:
Yes I checked all the configurations it is correct, I can attach samples!

Reply:
You also need to pass `use_polygons` & adapt the transforms accordingly to this `train_set` as well I think